### ISSUE TITLE
[BUG] Select widget has no value if only options are provided but the UI suggests otherwise

### ISSUE THREAD

**Issue by muendlein** [bokeh/bokeh#12088]
### Software versions

Python version : 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]
IPython version : (not installed)
Tornado version : 6.1
Bokeh version : 2.4.2

### Expected behavior

When using the Select widget and only providing options but no initial value, the UI field should either be completely empty or a default value is randomly selected (both UI + data).

### Observed behavior

The UI shows the first value of the options list however this value cannot be retrieved via the value property. 
I am not sure whether this intentional or not but I think this is a mismatch between data and view.

### Example code

```Python
from bokeh.layouts import row, column
from bokeh.models import Button, Select
from bokeh.plotting import curdoc, figure

doc = curdoc()

s = Select(options=["a","b","c"])

def show_selection():
    print(s.value) # prints/returns nothing

button = Button(label="select")
button.on_click(show_selection)

doc.add_root(column(button, s))
```

**Comment by bryevdv** [comment]
> When using the Select widget and only providing options but no initial value, the UI field should either be completely empty or a default value is randomly selected (both UI + data).

Noting that I believe the first option is the correct solution. "Nothing chosen" is a common initial starting configuration for select widgets.

**Comment by mattpap** [comment]
Currently the correct setup is to use `Select(options=["", "a", "b", "c"])` (or `None` instead of `""`). However, we need to al least add validation for `value` or remove its defaults value (`""` or `None`) or redesign `Select` widget  otherwise.

**Comment by mattpap** [comment]
This was fixed in PR #13497. A select with value property unset looks like this:
![image](https://github.com/bokeh/bokeh/assets/27475/c2ca28dd-676a-446c-92e7-b40edaa7f73d)
